### PR TITLE
fix: Correct the behaviour of Stripe endpoint and add relationship with Event

### LINKED ISSUE #4805: Wrong behaviour of the Stripe Authorization endpoint
**Describe the bug**
The Stripe Authorization endpoint currently takes the following parameters:

Parameter | Type | Required
-- | -- | --
stripe_secret_key | string | yes
stripe_refresh_token | string | yes
stripe_publishable_key | string | yes
stripe_user_id | string | yes
stripe_email | string | yes

This is not the correct behaviour. It leaves the work on getting all the above parameters on the Frontend.

**Expected behavior**
The Endpoint should accept the ```code``` from the Frontend. It should then use the client_secret stored in the settings model to get the other required information mentioned above. 
Moreover it should never return the stripe_secret_key parameter.

**Additional context**
Parent issue: #4796

### PR DESCRIPTION
**Pull request by srv-twry**
Fixes #4805 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Fixes the wrong behaviour of the Stripe Authorization endpoint. 
Endpoint takes in the ```code``` and uses it to fetch other details like ```secret_key``` from Stripe. 
Adds one-one relationship with Event

#### Changes proposed in this pull request:
1. Deleted fields from schema which aren't required.
2. Made stripe_publishable_key Read only.
3. Made code Write only.
4. Added relationship with Event.
5. Disallows two StripeAuthorization objects on one event.
6. Disallows Stripe Authorization object if stripe payment is disabled.
7. Fetches the other details from Stripe using the code.
8. Updated the docs.

### CODE REVIEW

**Comment by srv-twry** [comment]
TODO:

- [ ] Fix the build by updating the documentation.
- [ ] Use the code to fetch the details from Stripe

**NOT READY FOR REVIEW YET**

File: app/models/stripe_authorization.py [hunk 1]
```diff
@@ -14,7 +14,7 @@ class StripeAuthorization(db.Model):
     stripe_refresh_token = db.Column(db.String)
     stripe_publishable_key = db.Column(db.String)
     stripe_user_id = db.Column(db.String)
-    stripe_email = db.Column(db.String)
+    stripe_auth_code = db.column(db.String)
```
Comment (schedutron) [inline app/models/stripe_authorization.py]: Is this a typo? Shouldn't it be `db.Column`?

Comment (srv-twry) [reply app/models/stripe_authorization.py]: Yes,
Would have found out while migrating the db but thanks :)

**Comment by srv-twry** [comment]
Flow:
1. The Frontend sends the ```auth_code``` to the server by making a POST request.
2. The Server uses that ```code``` and the ```client_id``` of the admin from the settings to fetch the other details of the event organiser and saves it in the db. Frontend gets the ```stripe_publishable_key``` back. Other fields aren't required by the frontend hence aren't exposed.

Test:
ezgif com-video-to-gif


@niranjan94 @SaptakS @poush Please review :)

**Comment by srv-twry** [comment]
@mayank8318 @schedutron @bhaveshAn @Kreijstal Please review :)

**Comment by srv-twry** [comment]
@bhaveshAn Can you please tell me how to share a demo example here? I have added my docs there but can't find any option to share it.

**Comment by bhaveshAn** [comment]
@srv-twry  After creating doc goto Documentation and share that link.

**Comment by srv-twry** [comment]
https://openeventserver.docs.apiary.io/#

**Comment by srv-twry** [comment]
@bhaveshAn @mayank8318 @schedutron please review.

File: app/api/helpers/payment.py [hunk 2]
```diff
@@ -66,6 +72,27 @@ def get_credentials(event=None):
...
+    def get_event_organizer_credentials_from_stripe(stripe_auth_code):
+        """
+        Uses the stripe_auth_code to get the other credentials for the event organizer's stripe account
+        :param stripe_auth_code: stripe authorization code
+        :return: response from stripe
+        """
+        credentials = StripePaymentsManager.get_credentials()
+
+        if not credentials:
+            raise Exception('Stripe is incorrectly configured')
```
Comment (Kreijstal) [inline app/api/helpers/payment.py]: You should probably add a more meaningful error message, say that the credentials were not found.

Comment (bhaveshAn) [reply app/api/helpers/payment.py]: I agree

**Comment by srv-twry** [comment]
@bhaveshAn @Kreijstal @mayank8318 @schedutron Please review. 
PS: I know the build is failing but it is not failing due to this PR. You can review the changes since it will take time. Approve it when the build is fixed.

**Comment by srv-twry** [comment]
@bhaveshAn @mayank8318 @schedutron @Kreijstal Please review.

File: tests/hook_main.py [hunk 3]
```diff
@@ -3342,10 +3355,7 @@ def stripe_authorization_post(transaction):
     :param transaction:
     :return:
     """
-    with stash['app'].app_context():
-        event = EventFactoryBasic()
-        db.session.add(event)
-        db.session.commit()
+    transaction['skip'] = True
```
Comment (mayank8318) [inline tests/hook_main.py]: Why change this?

**Comment by srv-twry** [comment]
> maybe try to make PR smaller so it's easier to review them but LGTM right now

@Kreijstal Thanks for the review. Actually it wasn't possible here. 
I had to fix a few things(100 lines or so).Then the rest part(400 lines) was just updating the docs and routes otherwise the build would have failed.

**Pull request merged**